libraries/AP_InertialSensor/AP_InertialSensor.cpp

```diff
@@ -1262,10 +1262,10 @@ void AP_InertialSensor::update(void)
         bool have_zero_accel_error_count = false;
         bool have_zero_gyro_error_count = false;
         for (uint8_t i=0; i<INS_MAX_INSTANCES; i++) {
-            if (_accel_healthy[i] && _accel_error_count[i] <= _accel_startup_error_count[i]) {
+            if (_accel_healthy[i] && _accel_startup_error_count[i] != 0) {
```

Comment:
I don't think this change is correct.

The boolean being set indicates if we have a zero-error count from this sensor at this moment - after taking into account that we may get errors at start up (stored in the `_accel_startup_error_count` array).  You're saying here "if we had errors at startup then we have zero errors", which is not right.

I do think the code here is odd.  We should just be setting the primary instance of each sensor to the instance with the lowest error count, and not futzing around with the _healthy boolean at all.  To do so safely would require us to avoid integer overflow in the `_accel_error_count[]` flags etc.

Reply:
Peter is correct.

There were occasional issues with Solo coming up with bad IMU health. I observed that the error count was incremented during initialization and then never again thereafter. So, for lack of a better solution, I added _accel_startup_accel_count and initialized it some period of time after initialization. Thus, if there are IMU errors during initialization, they will be ignored.